results/client/client.go

```diff
@@ -0,0 +1,73 @@
+package client
+
+import (
+	"context"
+	"github.com/google/uuid"
+	proto "github.com/macformula/hil/results/client/generated"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const addr = "localhost:31763"
+
+type ResultsClient struct {
+	conn   *grpc.ClientConn
+	client proto.TagTunnelClient
+}
+
+func (r *ResultsClient) Open(ctx context.Context) error {
+	conn, err := grpc.DialContext(ctx, addr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		return err
+	}
+	r.conn = conn
+	r.client = proto.NewTagTunnelClient(conn)
+	return nil
+}
+
+func (r *ResultsClient) SubmitTag(ctx context.Context, tag string, value any) (bool, error) {
+	request, err := createRequest(tag, value)
+	if err != nil {
+		return false, err
+	}
+
+	reply, err := r.client.SubmitTag(ctx, request)
+	if err != nil {
+		return reply.Success, err
+	}
```

Comment:
Here you should use `reply.Success` to check the error message inside the reply. It's basically there to tell if the tag submitted actually exists. Keep the error check above as well though. 
```go
	if !reply.Success {
	    return false, errors.New(reply.Error)
	}
```